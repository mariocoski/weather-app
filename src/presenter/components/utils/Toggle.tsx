import * as React from 'react';
import { Checkbox } from 'semantic-ui-react';

interface IOptions {
  readonly defaultValue: boolean;
  readonly label: string;
  readonly onChange: (state: boolean) => void; 
}

const CheckboxExampleToggle = ({onChange, defaultValue, label} : IOptions) => (
  <Checkbox 
    label={label}
    toggle={true}
    // tslint:disable-next-line:jsx-no-lambda 
    onChange={(
      // tslint:disable-next-line:variable-name
      _e, data
    ) => {
      const state = data.checked !== undefined ?
      data.checked : defaultValue;
      onChange(state);
    }}
    defaultChecked={defaultValue}
  />
)

export default CheckboxExampleToggle