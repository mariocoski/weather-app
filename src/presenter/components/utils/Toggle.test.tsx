
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Toggle from './Toggle';

it('can toggle', () => {
  const handler = jest.fn();
  const component = renderer.create(
   <Toggle 
     defaultValue={true}
     onChange={handler}
     label={'label'}
   />,
  );

  const tree: any = component.toJSON();
 
  expect(tree).toMatchSnapshot();

  tree.props.onClick();

  expect(handler).toHaveBeenCalledTimes(1);
});
