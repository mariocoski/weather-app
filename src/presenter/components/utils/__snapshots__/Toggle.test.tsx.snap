// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can toggle 1`] = `
<div
  className="ui checked toggle checkbox"
  onChange={[Function]}
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <input
    checked={true}
    className="hidden"
    id={undefined}
    name={undefined}
    onClick={[Function]}
    readOnly={true}
    tabIndex={0}
    type="checkbox"
    value={undefined}
  />
  <label
    htmlFor={undefined}
  >
    label
  </label>
</div>
`;
